functions/quality_assessment/quality_assessment.py

```diff
@@ -145,12 +152,41 @@ def generate_message(bedrock_runtime: boto3.client,
     return response_body
 
 
-def get_prompt_list() -> list[dict]:
+@app.post("/update-prompt")
+@tracer.capture_method
+def update_prompt() -> dict:
+    """
+    A function to take in a prompt and prompt-id and update that prompt in the DynamoDB table
+
+    :returns: A simple status code dictionary
+    :raises Exception: raises an Exception if the DynamoDB write fails
+    """
+
+    body = app.current_event.json_body
+    try:
+        prompt_id = body["promptView"]["label"]
+        prompt_text = body["promptView"]["value"]
+
+        DYNAMO_CLIENT.put_item(TableName=TABLE_NAME,
+                                Item={
+                                    'prompt-id': {'S': prompt_id},
+                                    'prompt': {'S': prompt_text}
+                                })
+
+        return {"statusCode": 200}
+    except Exception as e:
+        logging.error(e)
+        return {"statusCode": 500, 
+                "body": e}
```

Comment:
It's probably ok for a sample application, but it would be better here to return a more generic error to the user, rather than the actual error message.
You might want to define the possible types of error (invalid input, internal error), and return a generic message based upon the type of error.